content/apply.js

```diff
@@ -42,24 +43,37 @@ self.INJECTED !== 1 && (() => {
     window.addEventListener(orphanEventId, orphanCheck, true);
   }
 
-  let parentDomain;
-
-  prefs.subscribe(['disableAll'], (key, value) => doDisableAll(value));
-  if (IS_FRAME) {
-    prefs.subscribe(['exposeIframes'], updateExposeIframes);
-  }
-
   function onInjectorUpdate() {
     if (!isOrphaned) {
       updateCount();
       updateExposeIframes();
+      const onOff = prefs[styleInjector.list.length ? 'subscribe' : 'unsubscribe'];
+      onOff(['disableAll'], updateDisableAll);
+      if (IS_FRAME) onOff(['exposeIframes'], updateExposeIframes);
+    }
+  }
+
+  async function init() {
+    if (STYLE_VIA_API) {
+      await API.styleViaAPI({method: 'styleApply'});
+    } else {
+      const styles = chrome.app && getStylesViaXhr() || await API.getSectionsByUrl(getMatchUrl());
+      await styleInjector.apply(styles);
     }
   }
 
-  function init() {
-    return STYLE_VIA_API ?
-      API.styleViaAPI({method: 'styleApply'}) :
-      API.getSectionsByUrl(getMatchUrl()).then(styleInjector.apply);
+  function getStylesViaXhr() {
+    if (new RegExp(`(^|\\s|;)${chrome.runtime.id}=\\s*([-\\w]+)\\s*(;|$)`).test(document.cookie)) {
+      const url = 'blob:' + chrome.runtime.getURL(RegExp.$2);
```

Comment:
I don't understand the syntax here. Does `RegExp` store the last matching result in its static property?

Reply:
Yes, that's how it always worked, and only recently this behavior was standardized :-)